### PR TITLE
Clarify lifecycle of monitoring exporter integration

### PR DESCRIPTION
**Pull request by rjeberhard**

### CODE REVIEW

File: documentation/domains/Domain.md [hunk 1]
```diff
@@ -37,7 +37,7 @@ The specification of the operation of the WebLogic domain. Required.
...
-| `monitoringExporter` | [Monitoring Exporter Specification](#monitoring-exporter-specification) | Configuration for the use of the WebLogic Monitoring Exporter as part of this domain. |
+| `monitoringExporter` | [Monitoring Exporter Specification](#monitoring-exporter-specification) | Automatic deployment and configuration of the WebLogic Monitoring Exporter. If specified, the operator will deploy a sidecar container alongside each WebLogic server instance that runs the exporter. WebLogic Server instances that are already running when the `monitoringExporter` field is created or deleted will not be restarted simply to provision or remove the exporter's sidecar container. When any given server is restarted for another reason, such as a change to the `restartVersion`, then the newly created pod will have the exporter sidecar or not, as appropriate. See https://github.com/oracle/weblogic-monitoring-exporter. |
```
Comment (rosemarymarano) [inline documentation/domains/Domain.md:40]: WebLogic server -> WebLogic Server
created or deleted -> created or deleted, (comma)

**Review by rosemarymarano — APPROVED** [review]
Minor nits, otherwise fine.

**Comment by tbarnes-us** [comment]
Related note: 

IMO, it'd be helpful if the https://github.com/oracle/weblogic-monitoring-exporter documentation directly cross-referenced this new related information in the WKO doc. There are two places an x-ref would help: the introduction bullet, and the 'Sidecar' section's intro (which should be expanded to indicate that the sidecar feature is primarily intended to help with WKO integration).

**Comment by rjeberhard** [comment]
@tbarnes-us, I'm going to work on some simple monitoring exporter integration documentation next (for WKO) and then I'll have a more appropriate page to link from the monitoring exporter site.

**Review by ddsharpe — APPROVED** [review]

**Pull request merged**